src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Conditions/ConditionTreeItem.cs

```diff
@@ -126,6 +127,13 @@ private void OnConditionUpdated(object sender, EventArgs e)
 
         public abstract bool CalculateResult(object entity);
 
+        /// <summary>
+        ///   Converts <see cref="ConditionTreeItem"/> to a LINQ <see cref="Expression"/>
+        /// </summary>
+        /// <param name="parameterExpr">LINQ ParameterExpression.</param>
+        /// <returns>LINQ Expression.</returns>
+        public abstract Expression ToLinqExpression(Expression parameterExpr);
```

Comment:
I would probably combine all ToLinqExpresison() methods and move it into the Extensions class or some "ToLiinqExpressionConverter"
not sure, but it looks and sounds like extension and looks like it is not really needed to put all this logic in separate classes